packages/ui/src/components/dropdown-menu.tsx

```diff
@@ -1,199 +1,438 @@
+  </div>
+)
+
+interface DropdownMenuItemProps
+  extends Omit<DropdownBaseItemProps, 'withSubIndicator'>,
+    Omit<ComponentPropsWithoutRef<typeof DropdownMenuPrimitive.Item>, 'title' | 'prefix'> {
+  prefix?: ReactNode
+}
+
+const DropdownMenuItem = forwardRef<ElementRef<typeof DropdownMenuPrimitive.Item>, DropdownMenuItemProps>(
+  ({ className, children, title, description, label, shortcut, checkmark, prefix, ...props }, ref) => {
+    const filteredChildren = filterChildrenByDisplayNames(children, innerComponentsDisplayNames)
+    const withChildren = filteredChildren.length > 0
+
+    const ItemContent = () => (
+      <DropdownBaseItem {...{ title, description, label, shortcut, checkmark, withSubIndicator: withChildren }}>
+        {prefix}
+      </DropdownBaseItem>
+    )
+
+    if (withChildren) {
+      return (
+        <DropdownMenuSub>
+          <DropdownMenuSubTrigger
+            ref={ref}
+            className={cn('cn-dropdown-menu-item cn-dropdown-menu-item-subtrigger', className)}
+            {...omit(props, ['onSelect'])}
+          >
+            <ItemContent />
+          </DropdownMenuSubTrigger>
+          <DropdownMenuContent isSubContent>{filteredChildren}</DropdownMenuContent>
+        </DropdownMenuSub>
+      )
+    }
+
+    return (
+      <DropdownMenuPrimitive.Item ref={ref} className={cn('cn-dropdown-menu-item', className)} {...props}>
+        <ItemContent />
+      </DropdownMenuPrimitive.Item>
+    )
+  }
+)
+DropdownMenuItem.displayName = displayNames.item
+
+interface DropdownMenuCheckboxItemProps
+  extends Omit<DropdownBaseItemProps, 'withSubIndicator'>,
+    Omit<ComponentPropsWithoutRef<typeof DropdownMenuPrimitive.CheckboxItem>, 'title' | 'onSelect'> {}
+
+const DropdownMenuCheckboxItem = forwardRef<
+  ElementRef<typeof DropdownMenuPrimitive.CheckboxItem>,
+  DropdownMenuCheckboxItemProps
+>(
+  (
+    {
+      className,
+      children,
+      checked,
+      title,
+      description,
+      label,
+      shortcut,
+      checkmark,
+      onCheckedChange,
+      onClick,
+      ...props
+    },
+    ref
+  ) => {
+    const filteredChildren = filterChildrenByDisplayNames(children, innerComponentsDisplayNames)
+    const withChildren = filteredChildren.length > 0
+    const checkedDataState = checked === true ? 'checked' : checked
+
+    const ItemContent = () => (
+      <DropdownBaseItem {...{ title, description, label, shortcut, checkmark, withSubIndicator: withChildren }}>
+        <div className="cn-checkbox-root" {...{ 'data-state': checkedDataState }}>
+          {checked && (
+            <div className="cn-checkbox-indicator" {...{ 'data-state': checkedDataState }}>
+              {checked === 'indeterminate' ? (
+                <Icon name="minus" className="cn-checkbox-icon" skipSize />
+              ) : (
+                <Icon name="check" className="cn-checkbox-icon" skipSize />
+              )}
+            </div>
```

Comment:
can we use the `<Checkbox />` component instead of a div with icons?

Reply:
Radix's checkboxes and radios are buttons with `aria-labels` under the hood. Having a clickable element inside a clickable menu item can lead to problems with focus when using keyboard navigation, which I encountered while working on the component.

<img width="310" alt="image" src="https://github.com/user-attachments/assets/b73045fa-3d76-4df2-9876-3cf3f6963b6b" />